cmsis-core/cmsis_nvic.h

```diff
@@ -34,6 +34,12 @@
 
 #include "cmsis.h"
 
+/* throw meaningful compiler error if yotta configuration is not provided */
+#if !defined(YOTTA_CFG_CMSIS_NVIC_USER_IRQ_OFFSET)      || !defined(YOTTA_CFG_CMSIS_NVIC_USER_IRQ_NUMBER) || \
```

Comment:
`core_generic.h` is a bit of a weird header (also, it's not really part of CMSIS, it was added by us :) ). It only includes the parts of CMSIS that are completely MCU independent. Most people will want to include `cmsis.h`, since that gives the whole package (generic part + MCU specific part).